toolchest_client/api/download.py

```diff
@@ -0,0 +1,127 @@
+"""
+toolchest_client.api.download
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This module provides an interface to download files contained on
+the Toolchest server, given access to an API key.
+
+Note: This module is used in downloading from the Output and
+Query classes.
+"""
+
+import os
+
+import boto3
+from botocore.exceptions import ClientError
+import requests
+from requests.exceptions import HTTPError
+
+from toolchest_client.api.auth import get_headers
+from toolchest_client.api.exceptions import ToolchestDownloadError
+from toolchest_client.api.urls import PIPELINE_URL
+from toolchest_client.files import get_file_type, get_params_from_s3_uri, unpack_files
+
+
+def download(output_path, s3_uri=None, pipeline_segment_instance_id=None,
+             output_file_keys=None, output_type=None, ):
+    """Downloads output to ``output_path``.
```

Comment:
Possible nit: unintentional "``"?

Reply:
That was a holdover from the previous comment in `query.py`, but it does seem unnecessary, so I'll get rid of it.